### DISCUSSION TITLE
Command launcher

### DISCUSSION THREAD

**Discussion by mozman** [mozman/ezdxf#331] [Ideas]
Easy way to launch ezdxf commands:

    python3 -m ezdxf cmd [options]

First commands:
- `pp` - the DXF pretty printer, replacement for the current `dxfpp` console script
- `audit` - audit and recover DXF files
- `draw` - current `draw_cad.py` example, use the matplotlib backend to convert DXF files
- `view` - current `cad_view.py` example, use the PyQt backend to view DXF files

Adding a console script `ezdxf`, I guess this will not collide with any Linux or macOS command:

    ezdxf cmd [options]

**Comment by danielkrajnik** [comment]
This is extremely useful - would it make sense to add it to the main README ? I didn't know that's what "Launcher" meant in the documentation. I was just looking for command line interface (fortunately title of this thread is **command** launcher)

**Reply by mozman** [reply]
I warmly invite your assistance in writing a concise paragraph that captures the essence of the feature for new users. As someone deeply involved in the project, it's become difficult for me to fully grasp the needs of newcomers. Your fresh perspective would be immensely valuable. Moreover, there is a wealth of documentation awaiting enhancements in a similar vein. Join us in making the project more accessible and user-friendly!

**Reply by danielkrajnik** [reply]
I was thinking of copying everything above up to `- view` and adding it as a new "Command Line" section. Would that be okay? I can submit it as a pull request.

**Reply by mozman** [reply]
That's fine, but the list of sub-commands has grown since this post was written: https://ezdxf.mozman.at/docs/launcher.html

**Reply by danielkrajnik** [reply]
Submitted pull request #908 - as a user I found only these three commands work well on my end. There are two commands to view .dxf files (view/draw), but matplotlib (`draw`) seemed much leaner than Qt backend (although a bit slower), so I included it instead of `view`. 

Please let me know if you think that's okay or if you'd prefer to include all the other commands as well.

**Reply by mozman** [reply]
Thank you.
